src/Dependencies/PooledObjects/ArrayBuilder.cs

```diff
@@ -563,5 +581,28 @@ public ImmutableArray<S> SelectDistinct<S>(Func<T, S> selector)
             set.Free();
             return result.ToImmutableAndFree();
         }
+
+        #region IDisposable Support
+        internal struct ArrayBuilderDisposer : IDisposable
+        {
+            private bool _disposed;
```

Comment:
>_disposed [](start = 25, length = 9)

Instead of the flag I'd make `_pooledObjects` writable and assign it null to indicate that it's been disposed.